repo/tuf_on_ci/build_repository.py

```diff
@@ -3,14 +3,95 @@
 """Command line tool to build a publishable TUF-on-CI repository"""
 
 import logging
+import os
+import subprocess
+from datetime import UTC, datetime, timedelta
+from urllib import parse
 
 import click
+from tuf.api.metadata import Root, Signed, Targets
 
 from tuf_on_ci._repository import CIRepository
+from tuf_on_ci._version import __version__
 
 logger = logging.getLogger(__name__)
 
 
+def _git(cmd: list[str]) -> subprocess.CompletedProcess:
+    cmd = [
+        "git",
+        "-c",
+        "user.name=tuf-on-ci",
+        "-c",
+        "user.email=41898282+github-actions[bot]@users.noreply.github.com",
+        *cmd,
+    ]
+    proc = subprocess.run(cmd, check=True, capture_output=True, text=True)
+    logger.debug("%s:\n%s", cmd, proc.stdout)
+    return proc
+
+
+def build_description(repo: CIRepository) -> str:
+    lines = [
+        "## TUF Repository state",
+        "",
+        "| Role | Next signing | Signers |",
+        "| - | - | - |",
+    ]
+    root = repo.root()
+    targets = repo.targets()
+    roles: list[tuple[Root | Targets, str]] = [
+        (root, "root"),
+        (root, "timestamp"),
+        (root, "snapshot"),
+        (root, "targets"),
+    ]
+    if targets.delegations and targets.delegations.roles:
+        for rolename in targets.delegations.roles:
+            roles.append((targets, rolename))
+
+    for delegator, rolename in roles:
+        role = delegator.get_delegated_role(rolename)
+        keys = [delegator.get_key(keyid) for keyid in role.keyids]
+        signers = []
+        for key in keys:
+            if "x-tuf-on-ci-keyowner" in key.unrecognized_fields:
+                signers.append(key.unrecognized_fields["x-tuf-on-ci-keyowner"])
+            else:
+                signers.append("_online key_")
```

Comment:
matter of taste, but this could be: 

```suggestion
            keyowner = key.unrecognized_fields.get("x-tuf-on-ci-keyowner", "_online key_")
            signers.append(keyowner)
```